composer/callbacks/speed_monitor.py

```diff
@@ -63,106 +63,106 @@ class SpeedMonitor(Callback):
     """
 
     def __init__(self, window_size: int = 100):
-        super().__init__()
-        self.train_examples_per_epoch = 0
-
-        # log the total wall clock time that this program has been running for
-        self.wall_clock_total = 0.0
+        # Track the epoch num samples and wct to compute throughput over the entire epoch
+        self.epoch_start_num_samples = Time.from_sample(0)
+        self.epoch_start_wct = datetime.timedelta(0)
+
+        # Track the batch num samples and wct to compute throughput over a window of batches
+        self.batch_start_num_samples = Time.from_sample(0)
+        self.batch_start_wct = datetime.timedelta(0)
+        self.rolling_batch_times_and_num_samples: Deque[Tuple[datetime.timedelta,
```

Comment:
Let's simplify this to just `self.buffer`? The variable name is too long, makes the code hard to read.

Or alternatively, separate it out into a window buffer for wct and one for `num_samples` to mirror everything else here.

Reply:
I like the latter suggustion